classes/task/failed_task.php

```diff
@@ -31,6 +31,98 @@ public function get_name() {
         return get_string('taskfailed', 'logstore_xapi');
     }
 
+    private function extract_failed_events($limitnum) {
+        global $DB;
+        $conditions = null;
+        $sort = '';
+        $fields = '*';
+        $limitfrom = 0;
+        $extractedevents = $DB->get_records('logstore_xapi_failed_log', $conditions, $sort, $fields, $limitfrom, $limitnum);
+        return $extractedevents;
+    }
+
+    private function extract_events($limitnum) {
```

Comment:
if we add second parameter which is the table name or something similar to the function we can use only one function instead of extract_failed_events and extract_events.
I mean for example:
```
    protected function extract_events($limitnum=0, $type='failed') {
        global $DB;

            if ($type == 'failed') {
                $events = $DB->get_records('logstore_xapi_failed_log', null, null, null, null, $limitnum);
            } else {
               $events = $DB->get_records('logstore_xapi_log', null, null, null, null, $limitnum);
            }
        return  $events;
    }

```

Reply:
I have combined the methods into one as recommended but the parameters cannot be null, null, null, null. They must have suitable defaults because the limit is configurable and passed by parameter. For easy configuration I have kept the original explicit declarations.

 ```
   private function extract_events($limitnum=0, $type='log') {
        global $DB;

        $conditions = null;
        $sort = '';
        $fields = '*';
        $limitfrom = 0;

        if ($type == 'log') {
            $events = $DB->get_records('logstore_xapi_log', $conditions, $sort, $fields, $limitfrom, $limitnum);
        } else {
            $events = $DB->get_records('logstore_xapi_failed_log', $conditions, $sort, $fields, $limitfrom, $limitnum);
        }
        return $events;
    }
```

Reply:
if you rename $type variable to $table
that you can only write the following:
```
$events = array();
if ($table == 'logstore_xapi_log' || $table == 'logstore_xapi_failed_log') {
    $events = $DB->get_records($table, $conditions, $sort, $fields, $limitfrom, $limitnum);
}
return $events;
```

what do you think?

Reply:
and the call looks like for_example:
$this->extract_events('logstore_xapi_log', 5);

Reply:
readable, isn't it?

Reply:
The condition if($table == 'logstore_xapi_log' || $table == 'logstore_xapi_failed_log') is too long since we know the tables already. This would involve 2 string compares each time the method is called.

So we don't have to pass the table name and to keep the definition for the method the same for the default table I think I will keep the $type parameter instead.